lms/static/js/jquery.gradebook.js

```diff
@@ -2,106 +2,129 @@
-		maxScroll = $gradeTable.width() - $grades.width();
-		var targetLeft = clamp($gradeTable.position().left, -maxScroll, 0);
-		updateHorizontalPosition(targetLeft);
-		setShadows(targetLeft);
-	};
-
-	var updateHorizontalPosition = function(left) {
-		$gradeTable.css({
-			'left': left + 'px'
-		});
-	};
-
-	var highlightRow = function(e) {
-		$element.find('.highlight').removeClass('highlight');
-
-		var index = $(this).index();
-		$studentTable.find('tr').eq(index + 1).addClass('highlight');
-		$gradeTable.find('tr').eq(index + 1).addClass('highlight');
-	};
-
-	var filter = function(e) {
-		var term = $(this).val();
-		if(term.length > 0) {
-			$studentTable.find('tbody tr').hide();
-			$gradeTable.find('tbody tr').hide();
-			$studentTable.find('tbody tr:contains(' + term + ')').each(function(i) {
-				$(this).show();
-				$gradeTable.find('tr').eq($(this).index() + 1).show();
-			});
-		} else {
-			$studentTable.find('tbody tr').show();
-			$gradeTable.find('tbody tr').show();
-		}
-	}
-
-	$leftShadow.css('height', tableHeight + 'px');
-	$rightShadow.css('height', tableHeight + 'px');
-	$grades.append($leftShadow).append($rightShadow);
-	setShadows(0);
-	$grades.css('height', tableHeight);
-	$gradeTable.bind('mousedown', startDrag);
-	$element.find('tr').bind('mouseover', highlightRow);
-	$search.bind('keyup', filter);
-	$(window).bind('resize', updateWidths);
-}
+    "use strict";
+    var $body = $('body');
+    var $grades = $element.find('.grades');
+    var $studentTable = $element.find('.student-table');
+    var $gradeTable = $element.find('.grade-table');
+    var $search = $element.find('.student-search-field');
+    var $leftShadow = $('<div class="left-shadow"></div>');
+    var $rightShadow = $('<div class="right-shadow"></div>');
+    var tableHeight = $gradeTable.height();
+    var maxScroll = $gradeTable.width() - $grades.width();
+
+    var mouseOrigin;
+    var tableOrigin;
+
+    var startDrag = function(e) {
+        mouseOrigin = e.pageX;
+        tableOrigin = $gradeTable.position().left;
+        $body.css('-webkit-user-select', 'none');
+        $body.bind('mousemove', onDragTable);
+        $body.bind('mouseup', stopDrag);
+    };
+
+    /**
+     * - Called when the user drags the gradetable
+     * - Calculates targetLeft, which is the desired position 
+     *   of the grade table relative to its leftmost position, using:
+     *   - the new x position of the user's mouse pointer;
+     *   - the gradebook's current x position, and;
+     *   - the value of maxScroll (gradetable width - container width).
+     * - Updates the position and appearance of the gradetable.
+     */
+    var onDragTable = function(e) {
+        var offset = e.pageX - mouseOrigin;
+        var targetLeft = clamp(tableOrigin + offset, maxScroll, 0);
+        updateHorizontalPosition(targetLeft);
+        setShadows(targetLeft);
+    };
+
+    var stopDrag = function() {
+        $body.css('-webkit-user-select', 'auto');
```

Comment:
Sadly, caniuse.com reports that [the `user-select` property isn't implemented unprefixed in any browser](http://caniuse.com/#search=user-select), and every browser has their own prefix. Since [Open edX supports Chrome, Safari, Firefox, and Internet Explorer](http://edx.readthedocs.org/projects/edx-installing-configuring-and-running/en/named-release-cypress/front_matter/browsers.html), we'll need to duplicate this `-webkit-user-select` logic with the other browser prefixes: `-moz-user-select` and `-ms-user-select`.